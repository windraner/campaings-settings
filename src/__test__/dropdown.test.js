import React from 'react';
import Dropdown from '../components/main-content-section/tab-block/common/dropdown/dropdown';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  productsCatalogueList: [
    'Product catalogue1',
    'Product catalogue2',
    'Product catalogue3',
    'Product catalogue4',
  ],
  title: 'test1',
  value: 'test2',
  field: 'test3',
  stateHandler() {}
};

describe('<Dropdown />', () => {
  it('Check title prop', () => {
    const wrapper = shallow(<Dropdown {...mock} />);
    expect(wrapper.find('.campaigns-settings__input-title').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__input-title').text()).toBe('test1');
  });
  it('Check hint text', () => {
    const wrapper = shallow(<Dropdown {...mock} />);
    expect(wrapper.find('.campaigns-settings__dropdown-hint').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__dropdown-hint').text()).toBe('Be careful! Changing catalog will result in all existing product associations to be lost.');
  });
  it('Check input', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<Dropdown {...mock} stateHandler={onClickFunc} />);
    const input = wrapper.find('.campaigns-settings__select');
    expect(input.props().value).toBe('test2');
    input.simulate('change', { target: { value: 'foo' }});
    expect(onClickFunc).toBeCalled();
    expect(onClickFunc.mock.calls).toEqual([['test3', 'foo']]);
  });
  it('matches the snapshot hide', () => {
    const tree = shallow(<Dropdown {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
