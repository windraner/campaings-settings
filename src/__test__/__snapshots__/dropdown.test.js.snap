// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> matches the snapshot hide 1`] = `
<Fragment>
  <label
    className="campaigns-settings__input-title"
  >
    test1
  </label>
  <div
    className="campaigns-settings__select-container position-relative"
  >
    <select
      className="campaigns-settings__select w-100"
      onChange={[Function]}
      value="test2"
    >
      <option
        className="campaigns-settings__select-option"
        key="0"
      >
        Product catalogue1
      </option>
      <option
        className="campaigns-settings__select-option"
        key="1"
      >
        Product catalogue2
      </option>
      <option
        className="campaigns-settings__select-option"
        key="2"
      >
        Product catalogue3
      </option>
      <option
        className="campaigns-settings__select-option"
        key="3"
      >
        Product catalogue4
      </option>
    </select>
    <span
      className="campaigns-settings__dropdown-icon"
    />
  </div>
  <div
    className="campaigns-settings__dropdown-hint mt-2 position-relative"
  >
    <span
      className="campaigns-settings__dropdown-hint-icon"
    />
    Be careful! Changing catalog will result in 
    <strong>
      all existing product associations to be lost.
    </strong>
  </div>
</Fragment>
`;
